index.js

```diff
@@ -29,31 +31,53 @@ const run = config => {
     });
 };
 
-const checkDirectoryStatus = dest => {
-    return fse.stat(dest)
-        .then(statResp => {
-            if (!statResp.isDirectory) {
-                return fse.mkdir(dest);
-            }
-            return Promise.resolve();
-        })
-        .catch(() => fse.mkdir(dest));
-};
-
-const copyDirectory = fileConfig => {
-    return checkDirectoryStatus(fileConfig.dest).then(() => {
-        return fse.readdir(fileConfig.src)
-            .then(files => {
-                const filePromises = files.map(file => copyFile(fileConfig, file));
-                return Promise.all(filePromises);
-            });
+/**
+ * 
+ * @param {String} srcPath the full src path from the config
+ * This function finds the base path which is the path before any glob syntax is used
+ * This helps parse the file path in the copyFile method
+ * @returns {String} base path (the path before any glob syntx is used)
+ */
+function getSourceDir(srcPath) {
+    const srcPaths = srcPath.split('/');
+    let retVal = srcPaths;
+    srcPaths.forEach((path, indx) => {
+        if (!glob.hasMagic(path)) {
+            const endIndx = indx + 1;
+            retVal = srcPaths.slice(0, endIndx).join('/');
+        }
     });
+    return retVal;
+}
+
+/**
+ *
+ * @param {Object} fileConfig config object from project
+ * this method gets all files in src (taking into account globbing syntax)
+ * it then calls the copyFile method to copy each file in directory
+ * @returns {Promise} a promise of all copyFile promises
+ */
+const copyDirectory = async (fileConfig) => {
+    fileConfig.basePath = getSourceDir(fileConfig.src);
+    const filePaths = await glob(`${fileConfig.src}`);
```

Comment:
This seems like an unnecessary template string. Can you just pass the variable directly?